ts/components/messages/MessageAgenda.tsx

```diff
@@ -252,82 +243,148 @@ const generateItemLayouts = (sections: Sections) => {
+    <View style={styles.itemLoadingHeaderWrapper}>
+      <View style={styles.itemLoadingHeaderCenter}>
+        <Placeholder.Paragraph
+          textSize={customVariables.fontSizeBase}
+          color={customVariables.shineColor}
+          lineNumber={2}
+          lineSpacing={5}
+          width="100%"
+          firstLineWidth="100%"
+          lastLineWidth="55%"
+          animate="shine"
+          onReady={false}
+        />
+      </View>
+    </View>
+
+    <View style={styles.itemLoadingContentWrapper}>
+      <View style={styles.itemLoadingContentCenter}>
+        <Placeholder.Line
+          textSize={customVariables.fontSizeBase}
+          color={customVariables.shineColor}
+          width="75%"
+          animate="shine"
+        />
+      </View>
+    </View>
   </View>
 );
 
+const SectionHeaderPlaceholder = (
+  <View style={styles.sectionHeaderWrapper}>
+    <View style={styles.sectionHeaderContent}>
+      <Placeholder.Line
+        textSize={customVariables.fontSizeBase}
+        color={customVariables.shineColor}
+        width="75%"
+        animate="shine"
+      />
+    </View>
+  </View>
+);
 /**
  * A component to render messages with due_date in a agenda like form.
  */
 class MessageAgenda extends React.PureComponent<Props, State> {
+  private idTimeout?: number;
+  private idInterval?: number;
+  private messageToLoadFromSections: number = 0;
   // Ref to section list
   private sectionListRef = React.createRef<any>();
-  private idTimeoutProgress?: number;
-
   constructor(props: Props) {
     super(props);
     this.state = {
       itemLayouts: [],
       isLoadingProgress: false,
-      isFirstLoading: true
+      isFirstLoading: true,
+      isDeadlinesLoaded: false,
+      isLoadingComplete: false,
+      numMessagesToRender: 0
     };
-    this.loadMoreData = this.loadMoreData.bind(this);
+  }
+
+  public componentDidMount() {
+    // tslint:disable-next-line: no-object-mutation
+    this.idInterval = setInterval(() => {
+      if (!this.state.isDeadlinesLoaded) {
+        this.isDeadlinesLoadingComplete();
+      } else {
+        clearInterval(this.idInterval);
+      }
+    }, 300);
   }
 
   public componentWillUnmount() {
-    if (this.idTimeoutProgress !== undefined) {
-      clearTimeout(this.idTimeoutProgress);
+    // if a timeout is running we have to stop it
+    if (this.idTimeout !== undefined) {
+      clearTimeout(this.idTimeout);
+    }
+    if (this.idInterval !== undefined) {
+      clearInterval(this.idInterval);
     }
   }
 
-  // tslint:disable-next-line: cognitive-complexity
-  public componentDidUpdate(prevProps: Props) {
-    // Change status loading to show progress
+  public async componentDidUpdate() {
     if (
-      prevProps.refreshing !== this.props.refreshing &&
-      this.props.refreshing === false
+      this.state.isDeadlinesLoaded &&
+      this.sectionListRef.current !== null &&
+      this.state.isFirstLoading &&
+      this.props.sections !== undefined &&
```

Comment:
Maybe the condition verifies that the item is not undefined if you write just:
```suggestion
      this.sectionListRef.current &&
      this.state.isFirstLoading &&
      this.props.sections &&
```
I'm less sure about null object, can you check it?